rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -1452,7 +1493,8 @@ rmw_create_client(
   // TODO(Yadunund): We cannot store the rmw_node_t * here since this type erased
   // Client handle will be returned in the rmw_clients_t in rmw_wait
   // from which we cannot obtain ClientData.
-  rmw_client->data = static_cast<void *>(node_data->get_client_data(rmw_client).get());
+  ClientDataPtr client_data = node_data->get_client_data(rmw_client);
```

Comment:
```suggestion
  rmw_zenoh_cpp::ClientDataPtr client_data = node_data->get_client_data(rmw_client);
```

Reply:
thanks. `rmw_zenoh_cpp` was failing to build on my (22.04) system due to something else and I couldn't figure out why.